crates/ruff_python_parser/src/python_version.rs

```diff
@@ -0,0 +1,314 @@
+use std::fmt;
+
+pub use pyversion::PyVersion;
+
+mod pyversion {
```

Comment:
I'd prefer to only have one python version in the parser crate. Having more than one seems confusing and unnecessary. Let's keep this in the linter if the linter isn't able to migrate to `PythonVersion` just yet. It makes it clear why the extra type exists and to which type we should migrate to